### PR TITLE
Add secondary NIC support requirement for Manila CSI

### PR DESCRIPTION
**Pull request by Fedosin**

### CODE REVIEW

**Comment by tombarron** [comment]
/lgtm -- I probably don't have rights for this here, but for the record :D

To deploy networked storage with OpenShift on OpenStack with isolated networks (as done standardly by TripleO or OSP director), or indeed with any OpenStack storage back end that does the smart thing and presents storage over a dedicated isolated storage network, this is necessary.  Please feel free to reach out and discuss OpenStack networking/storage topologies with me if you have any questions.

File: enhancements/storage/manila-csi-driver-operator.md [hunk 1]
```diff
@@ -142,6 +142,16 @@ Actions:
 
 The operator will run containerized and End-to-End tests and also push the resulting image in the OCP Quay account.
 
+#### Implement Secondary NIC support
+
+By default in OSP Manila uses additional network for volumes. This means that in addition to the main network, we must connect worker nodes to the appropriate storage network to be able to successfully mount the volumes.
+
+Actions:
+
+- Add the ability to specify additional networks and security groups in the installer configuration.
```
Comment (ashcrow) [inline enhancements/storage/manila-csi-driver-operator.md:151]: Is the installer team aware?

Comment (Fedosin) [reply enhancements/storage/manila-csi-driver-operator.md:151]: Yes! @abhinavdahiya knows about this requirement. https://github.com/openshift/installer/pull/3291

Comment (abhinavdahiya) [reply enhancements/storage/manila-csi-driver-operator.md:151]: can you include the expected design of these fields are going to be added here, from https://github.com/openshift/installer/pull/3291

and would like to see what validations etc. will be added to these new fields.

Comment (Fedosin) [reply enhancements/storage/manila-csi-driver-operator.md:151]: Done!

Comment (Fedosin) [reply enhancements/storage/manila-csi-driver-operator.md:151]: And regarding to validations: https://github.com/openshift/installer/pull/3291/commits/cf318fc3536ea51cd8832b495498d1eb8b1f82c4

**Comment by jsafrane** [comment]
lgtm from storage perspective - this is about network configuration which is out of our scope.

However, one question: is it possible to add the networks and security groups post-install? For example, after upgrading 4.x cluster to 4.x+1, I want to enable Manila. But the old cluster did not have extra storage network configured. Or, I forgot to add the networks during installation, how do I add them now?

Is it something we want to support or will we solve it by documented limitation? IMO the enhancement should have a short note about that.

**Comment by Fedosin** [comment]
@jsafrane sure! you can update machine configurations after the installation to add networks and security groups there.
But we need this in IPI because we plan to use Manila as a primary backend for Image Registry.

**Comment by Fedosin** [comment]
I'll update the patch soon

**Comment by Fedosin** [comment]
@jsafrane done :)

**Comment by abhinavdahiya** [comment]
LGTM

**Comment by mandre** [comment]
LGTM

**Pull request merged**